### PR TITLE
upgrade to FakeItEasy 2.0.0

### PR DESCRIPTION
**Pull request by adamralph**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/NancyFx/Nancy/pulls) open
- [x] I have verified that I am following the Nancy [code style guidelines](https://github.com/NancyFx/Nancy/blob/45238076ad0b7f6ecabd6bae8469e30458d02efe/CONTRIBUTING.md#style-guidelines)
- [x] I have provided test coverage for my change (where applicable)

### Description
As it says on the tin.

### CODE REVIEW

File: test/Nancy.Tests/Unit/ModelBinding/DefaultBodyDeserializers/JsonBodyDeserializerFixture.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public void Should_report_false_for_can_deserialize_for_non_json_format()
             const string contentType = "application/xml";
 
             // When
-            var result = this.deserialize.CanDeserialize(contentType, A<BindingContext>._);
+            var result = this.deserialize.CanDeserialize(contentType, A.Dummy<BindingContext>());
```
Comment (adamralph) [inline test/Nancy.Tests/Unit/ModelBinding/DefaultBodyDeserializers/JsonBodyDeserializerFixture.cs]:
```C#
A<BindingContext>._
```
is designed for assertion only (e.g. `A.CallTo()`) which does not execute the `_` property.

In FakeItEasy 2.0, if `_` is ever executed, an exception is thrown. [Dummies](http://fakeiteasy.readthedocs.io/en/2.0.0/dummies/) are the simplest constructs for this use case.

File: test/Nancy.Tests/Unit/Bootstrapper/NancyBootstrapperBaseFixture.cs [hunk 2]
```diff
@@ -131,17 +131,9 @@ public void Should_invoke_startup_tasks_after_registration_tasks()
...
-                // Then
-                using (scope.OrderedAssertions())
-                {
-                    A.CallTo(() => registrations.CollectionTypeRegistrations).MustHaveHappened();
-                    A.CallTo(() => startup.Initialize(A<IPipelines>._)).MustHaveHappened();
-                }
-            }
+            this.bootstrapper.Initialise();
+            A.CallTo(() => registrations.CollectionTypeRegistrations).MustHaveHappened()
+                .Then(A.CallTo(() => startup.Initialize(A<IPipelines>._)).MustHaveHappened());
```
Comment (adamralph) [inline test/Nancy.Tests/Unit/Bootstrapper/NancyBootstrapperBaseFixture.cs]: [Ordered assertions](http://fakeiteasy.readthedocs.io/en/2.0.0/ordered-assertions/) have been completely overhauled in FIE 2.0. Scopes no longer exist.

File: test/Nancy.Tests/Unit/Bootstrapper/NancyBootstrapperBaseFixture.cs [hunk 3]
```diff
@@ -131,17 +131,11 @@ public void Should_invoke_startup_tasks_after_registration_tasks()
...
-                {
-                    A.CallTo(() => registrations.CollectionTypeRegistrations).MustHaveHappened();
-                    A.CallTo(() => startup.Initialize(A<IPipelines>._)).MustHaveHappened();
-                }
-            }
+            this.bootstrapper.Initialise();
+
+            // Then
+            A.CallTo(() => registrations.CollectionTypeRegistrations).MustHaveHappened()
+                .Then(A.CallTo(() => startup.Initialize(A<IPipelines>._)).MustHaveHappened());
```
Comment (adamralph) [inline test/Nancy.Tests/Unit/Bootstrapper/NancyBootstrapperBaseFixture.cs]: [Ordered assertions](http://fakeiteasy.readthedocs.io/en/2.0.0/ordered-assertions/) have been completely overhauled in FIE 2.0. Scopes no longer exist.

File: test/Nancy.Tests.MSBuild/Nancy.Tests.csproj [hunk 4]
```diff
@@ -276,15 +276,15 @@
...
-    </Compile>
+    <Compile Include="..\Nancy.Tests\Unit\HttpLinkBuilderFixture.cs">
+      <Link>Unit\HttpLinkBuilderFixture.cs</Link>
+    </Compile>
+    <Compile Include="..\Nancy.Tests\Unit\HttpLinkFixture.cs">
+      <Link>Unit\HttpLinkFixture.cs</Link>
+    </Compile>
+    <Compile Include="..\Nancy.Tests\Unit\HttpLinkRelationFixture.cs">
+      <Link>Unit\HttpLinkRelationFixture.cs</Link>
+    </Compile>
```
Comment (adamralph) [inline test/Nancy.Tests.MSBuild/Nancy.Tests.csproj]: This was due to inconsistent line breaks. These lines had LF and rest of the file has CRLF.

I can remove this change from the commit if you prefer to keep it pure.

**Comment by khellang** [comment]
Thanks, @adamralph! :heart: Will this allow us to run the tests on netstandard?

**Comment by adamralph** [comment]
The platform hasn't changed. Still good old .NET 4.0.

**Comment by khellang** [comment]
Ah, saw https://github.com/FakeItEasy/FakeItEasy/pull/617 now :smile:

**Comment by adamralph** [comment]
@khellang it does allow this though - https://github.com/NancyFx/Nancy/pull/2490

**Comment by jchannon** [comment]
we want netstandard :trollface: :trollface:

**Comment by adamralph** [comment]
Here you go: https://github.com/jeremymeng/FakeItEasy/tree/coreclr

**Comment by jchannon** [comment]
:)

**Pull request merged**